.github/workflows/phpcs.yml

```diff
@@ -31,59 +35,134 @@ jobs:
         id: get-composer-cache-dir
         run: |
           echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
-      - uses: actions/cache@v2
-        id: composer-cache
+      - name: Cache Composer dependencies
+        uses: actions/cache@v2
         with:
           path: ${{ steps.get-composer-cache-dir.outputs.dir }}
           key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
           restore-keys: |
             ${{ runner.os }}-composer-
 
-      - uses: "shivammathur/setup-php@v2"
-        with:
-          php-version: ${{ inputs.php_version }}
-      - uses: "ramsey/composer-install@v2"
-
-      - name: "Give permissions"
+      # ------------------------------------------------------------------------------
+      # Check for PHPCS
+      # ------------------------------------------------------------------------------
+      - name: Check PHPCS installation
         run: |
-          sudo chown -R root:root $GITHUB_WORKSPACE
+          if [ ! -f vendor/bin/phpcs ]; then
+            echo "PHPCS not found in vendor/bin, installing..."
+            composer require "squizlabs/php_codesniffer=*"
+          else
+            echo "PHPCS found in vendor/bin"
+          fi
+          vendor/bin/phpcs --version
 
       # ------------------------------------------------------------------------------
       # Get changed files
       # ------------------------------------------------------------------------------
       - name: Get list of changed files
-        id: files
+        id: get-changed-files
         run: |
-          echo "CHANGED_FILES=$(git diff --name-only --diff-filter=AM ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} -- . ':!tests' | grep '\.php$' | tr '\n' ' ')" >> $GITHUB_ENV
+          CHANGED_FILES=$(git diff --name-only --diff-filter=AM ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} -- . ':!tests' | grep '\.php$')
+          echo "CHANGED_FILES<<EOF" >> $GITHUB_ENV
+          echo "${CHANGED_FILES}" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
```

Comment:
Something weird is going on here - i think we need L67 and 68 and not 66

Reply:
I just tested it with removing line 67. All three of the lines are required. This is due to feeding in the `git diff` as multiline instead of space deliminated.

Reply:
Alrighty, since it works im fine with that!